File: go-code/refactoring_api/replace_query_with_parameter/before/heating_plan.go
package before

type Thermostat struct {
	SelectedTemperature int
	CurrentTemperature int
}

type Plan struct {
	Min, Max   int
	Thermostat Thermostat
}

func (plan Plan) getTargetTemperature() int {
	if plan.Thermostat.SelectedTemperature > plan.Max {
		return plan.Max
	} else if plan.Thermostat.SelectedTemperature < plan.Min {
		return plan.Min
	} else {
		return plan.Thermostat.SelectedTemperature
	}
}

func main() {
	thermostat := Thermostat{SelectedTemperature: 25}
	thePlan := Plan{Min: 20, Max: 30, Thermostat: thermostat}

	if thePlan.getTargetTemperature() > thermostat.CurrentTemperature {
		setToHeat()
	} else if thePlan.getTargetTemperature() < thermostat.CurrentTemperature {
		setToCool()
	} else {
		setOff()
	}
}

func setOff() {

}

func setToCool() {

}

func setToHeat() {

}

File: java-code/src/main/java/refactoring/refactoring_api/replace_query_with_parameter/before/HeatingPlan.java
package refactoring.refactoring_api.replace_query_with_parameter.before;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class HeatingPlan {
  int min, max;
  Thermostat thermostat;

  int getTargetTemperature() {
    if (thermostat.currentTemperature > max) {
      return max;
    } else if (thermostat.currentTemperature < min) {
      return min;
    } else {
      return thermostat.currentTemperature;
    }
  }
}


